gitlab/resource_gitlab_project_environment.go

```diff
@@ -0,0 +1,163 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/ci/environments/protected_environments.html
+func resourceGitlabProjectEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectEnvironmentCreate,
+		Read:   resourceGitlabProjectEnvironmentRead,
+		Update: resourceGitlabProjectEnvironmentUpdate,
+		Delete: resourceGitlabProjectEnvironmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"external_url": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"state": {
```

Comment:
Can you add this attribute to the doc page

Reply:
Done in be501eb73c189081777f409af7cc2dd3b5ccc77a